server/src/main/java/org/opensearch/cluster/metadata/QueryGroup.java

```diff
@@ -0,0 +1,304 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.joda.time.Instant;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class to define the ResourceLimitGroup schema
+ * {
+ *     "fafjafjkaf9ag8a9ga9g7ag0aagaga" : {
+ *              "jvm": 0.4,
+ *              "mode": "enforced",
+ *              "name": "analytics",
+ *              "updatedAt": 4513232415
+ *      }
+ * }
+ */
+@ExperimentalApi
+public class QueryGroup extends AbstractDiffable<QueryGroup> implements ToXContentObject {
+
+    public static final int MAX_CHARS_ALLOWED_IN_NAME = 50;
+    private final String name;
+    private final String _id;
+    private final QueryGroupMode mode;
```

Comment:
Given the generalization to QueryGroup from Sandbox/ResourceLimitGroup, mode alone as an attribute is not very clear. Shall we call this cancellation or resiliency mode? Or anything else you can come up with

Reply:
Make sense!